File: dart_sample/lib/demos/cupertino/cupertino_search_text_field_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

class CupertinoSearchTextFieldDemoPage extends StatefulWidget {
  const CupertinoSearchTextFieldDemoPage({super.key});

  @override
  State<CupertinoSearchTextFieldDemoPage> createState() =>
      _CupertinoSearchTextFieldDemoPageState();
}

class _CupertinoSearchTextFieldDemoPageState
    extends State<CupertinoSearchTextFieldDemoPage> {
  final TextEditingController _controller = TextEditingController();
  String _query = '';

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'Cupertino search text field',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Localized placeholder, live clear button, custom icon treatment, '
          'and disabled state.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        CupertinoSearchTextField(
          controller: _controller,
          onChanged: (String value) => setState(() => _query = value),
        ),
        Text(
          _query.isEmpty ? 'No query' : 'Query: $_query',
          style: const TextStyle(fontSize: 12, color: Color(0xFF607D8B)),
        ),
        const CupertinoSearchTextField(
          placeholder: 'Always-visible action',
          itemColor: CupertinoColors.systemBlue,
          itemSize: 24,
          suffixMode: OverlayVisibilityMode.always,
        ),
        const CupertinoSearchTextField(
          placeholder: 'Disabled search',
          enabled: false,
        ),
      ],
    );
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }
}

File: src/Sample/Plumix.Sample/Demos/Cupertino/CupertinoSearchTextFieldDemoPage.cs
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/cupertino_search_text_field_demo_page.dart

public sealed class CupertinoSearchTextFieldDemoPage : StatefulWidget
{
    public override State CreateState() => new CupertinoSearchTextFieldDemoPageState();
}

internal sealed class CupertinoSearchTextFieldDemoPageState : State
{
    private readonly TextEditingController _controller = new();
    private string _query = string.Empty;

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12.0,
            children:
            [
                new Text("Cupertino search text field", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "Localized placeholder, live clear button, custom icon treatment, and disabled state.",
                    fontSize: 14.0,
                    color: Color.FromUInt32(0x8A000000)),
                new CupertinoSearchTextField(
                    controller: _controller,
                    onChanged: value => SetState(() => _query = value)),
                new Text(
                    string.IsNullOrEmpty(_query) ? "No query" : $"Query: {_query}",
                    fontSize: 12.0,
                    color: Color.FromUInt32(0xFF607D8B)),
                new CupertinoSearchTextField(
                    placeholder: "Always-visible action",
                    itemColor: CupertinoColors.SystemBlue,
                    itemSize: 24.0,
                    suffixMode: OverlayVisibilityMode.Always),
                new CupertinoSearchTextField(
                    placeholder: "Disabled search",
                    enabled: false),
            ]);
    }

    public override void Dispose()
    {
        _controller.Dispose();
        base.Dispose();
    }
}
